examples/dbscan/dbscan.cpp

```diff
@@ -60,8 +60,8 @@ std::vector<ArborX::Point> loadData(std::string const &filename,
   {
     v.reserve(num_points);
 
-    auto read_point = [&input, dim]() {
-      auto it = std::istream_iterator<float>(input);
+    auto it = std::istream_iterator<float>(input);
+    auto read_point = [&it, dim]() {
```

Comment:
I fail to see what was wrong before

Reply:
Me too. However, the read in data was completely wrong. Moreover, valgrind complained
```
=29827==  Uninitialised value was created by a stack allocation
==29827==    at 0x11562A: loadData(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, bool, int)::{lambda()#1}::operator()() const (dbscan.cpp:64)
```

Reply:
anyway no objection to the change either